File: server/biz/prm/tag.go
package prm

import (
	"onij/model/api"
	"onij/util"
)

type UploadTagParam struct {
	ResourceId   int64
	ResourceType int32
	TagBiz       int32
	TagGroup     int32
	TagType      int32
	TargetId     *int64
	TargetType   *int32
	Extra        string
}

func NewUploadTagParam(req *api.UploadTagReq) *UploadTagParam {
	return &UploadTagParam{
		ResourceId:   req.Tag.ResourceId,
		ResourceType: int32(req.Tag.ResourceType),
		TagBiz:       int32(req.Tag.TagBiz),
		TagGroup:     int32(req.Tag.TagGroup),
		TagType:      int32(req.Tag.TagType),
		TargetId:     req.Tag.TargetId,
		TargetType:   (*int32)(req.Tag.TargetType),
		Extra:        req.Tag.Extra,
	}
}

type UploadTagResult struct {
}

func (r *UploadTagResult) Resp() *api.UploadTagResp {
	return &api.UploadTagResp{
		Code:    util.BaseCodeOK,
		Message: util.BaseMsgOK,
	}
}

type DeleteTagParam struct {
	ResourceId int64
	TagType    int32
}

func NewDeleteTagParam(req *api.DeleteTagReq) *DeleteTagParam {
	return &DeleteTagParam{
		ResourceId: req.ResourceId,
		TagType:    int32(req.TagType),
	}
}

type DeleteTagResult struct{}

func (r *DeleteTagResult) Resp() *api.DeleteTagResp {
	return &api.DeleteTagResp{
		Code:    util.BaseCodeOK,
		Message: util.BaseMsgOK,
	}
}

File: src/api/types/tag.ts
import { TagBiz, TagGroup, TagType, ResourceType } from './enums'

export interface Tag {
  id: number
  resourceId: number
  resourceType: ResourceType
  tagBiz: TagBiz
  tagGroup: TagGroup
  tagType: TagType
  targetId: number
  extra: string
}

// 请求和响应类型
export interface UploadTagReq {
  tag: Tag
}

export interface UploadTagResp {
  code: number
  message: string
}

export interface DeleteTagReq {
  resourceId: number
  tagType: TagType
}

export interface DeleteTagResp {
  code: number
  message: string
}